lora_pkg_fwd

iot-floriani opened the pull request:
Hi, 

would like to modify the lora_pkg_fwd firmware, but even if I download all the files, I cannot compile it, due to errors and lack of libraries.

Somebody  can share all the necessary files with me?

Thank you

dragino committed:
add TCP_IP mode support

dragino committed:
add a single_tx utility for test sx1276

dragino committed:
reload module gpio_keys_polled can release the 12 gpio port

dragino committed:
add a test utility for sx1276

dragino committed:
fix the board detect ulitity

dragino committed:
fix invertIQ and optimize lower datarate

dragino committed:
add setpower function, add the downlink manual process

dragino committed:
add lg01 support, change check board applicate

dragino committed:
fix a bug of file check_board

dragino committed:
merge lg02 and lg01

dragino committed:
fix lg08 connot custome conconfigure sx130x, lg02 keealive for recieve continue mode

dragino committed:
fix: lg02 relay prevent circul, lg08 customer configure, disable mode

dragino committed:
1/ Fix sync word bug configure bug in lg02_single_rx_tx
2/ Add option to set Power in lg02_single_rx_tx
3/ Add RadioHead Support (-R option) in lg02_single_rx_tx

dragino committed:
add mqtt support

dragino committed:
fix lg01_pkt_fwd boot bug

dragino committed:
add custom tx power for LG308

dragino committed:
add debug info of txpower

dragino committed:
fix a bug

dragino committed:
fix bug read uci config

dragino committed:
fix bug about LG308 using SX1276 for TX

dragino committed:
fix the delay for useing sx1276 TX

dragino committed:
add delay for txlora when using sx1276

dragino committed:
add monitor interface status for fwd

dragino committed:
add pps info for lg08, add dir for lg02 receive

dragino committed:
Add customized script support

dragino committed:
add udp socket alive detect

dragino committed:
add udp socket alive detect

dragino committed:
Merge branch 'lg02' of github.com:dragino/dragino-packages into lg02

dragino committed:
fix a bug cannot set socke option

dragino committed:
fix bug about thread_push cannot start on mqtt mode

dragino committed:
fix a bug of DOWNLINK_CLASS_C

dragino committed:
add some option for lg02_single_rx_tx test suit

dragino committed:
update

dragino committed:
lorawan simple serv project

dragino committed:
add database control

mikayong committed:
add database struct

dragino committed:
update

dragino committed:
fix a bug in Makefiel

dragino committed:
update

dragino committed:
update

dragino committed:
update

dragino committed:
update

dragino committed:
fix some bug

mikayong committed:
update

mikayong committed:
update

mikayong committed:
merge

mikayong committed:
fix

mikayong committed:
update

mikayong committed:
add lora_serv a simple lorawan server

mikayong committed:
test

mikayong committed:
update

mikayong committed:
add a script for check netconnection

mikayong committed:
fix checkalive config to gateway

mikayong committed:
fix bug

mikayong committed:
add field for identify the msg if push

mikayong committed:
add utili import devinfo

dragino committed:
change to dls_utili.c filename

mikayong committed:
add some default profile, add mandev function to add device by manual

mikayong committed:
update

mikayong committed:
add import csv option

mikayong committed:
fix bug not effect on delete pfname

dragino committed:
update

mikayong committed:
add mqtt suport

mikayong committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

mikayong committed:
add lsnr report

mikayong committed:
add time field for rxpk

dragino committed:
upload code

dragino committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

mikayong committed:
RSSI correction

mikayong committed:
upadte

mikayong committed:
update

mikayong committed:
update

dragino committed:
mqtt improve

dragino committed:
fix rxpk format add metadata time

mikayong committed:
init cntx struct

mikayong committed:
modify a util_test_hal util for test loralib

mikayong committed:
change the method of pull data

dragino committed:
Add new GUI. improve mqtt process.

dragino committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

mikayong committed:
add mac header decode for debug message

mikayong committed:
add a count of fwdack for sync

dragino committed:
add dragino gateway packages

dragino committed:
udpate

dragino committed:
init resp

dragino committed:
update

dragino committed:
update

dragino committed:
update

dragino committed:
fix a mistake

dragino committed:
fix a mistake

dragino committed:
update

dragino committed:
update Makefile for mqtt

dragino committed:
update

dragino committed:
update

dragino committed:
update

dragino committed:
update

dragino committed:
update

dragino committed:
update

dragino committed:
update

dragino committed:
udpate

dragino committed:
udpate

dragino committed:
update

dragino committed:
init resp finish

dragino committed:
add output connetion status to /var/iot/status file

dragino committed:
gw forward update

dragino committed:
remove a gw forward on branch lg02

dragino committed:
add SIGUSR1 handle for reconnect the loraserver

dragino committed:
add signal handle sigusr

dragino committed:
update pkt_fwd

dragino committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

dragino committed:
use fport filter upload package

dragino committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

dragino committed:
indication when drop pacakage filter by fport

dragino committed:
Merge branch 'lg02' of github.com:dragino/dragino-packages into lg02

dragino committed:
export rssi to file

dragino committed:
1/fix lg01/lg02 rssi issue.
2/add Fport filter
3/change default gateway config settiings

dragino committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

dragino committed:
improve OEM structure

dragino committed:
add mosquitto pacakge in ulitis category

dragino committed:
output status on starup

dragino committed:
Add keep alive script

dragino committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

dragino committed:
Add Devaddr filter

dragino committed:
add maccrypto for abpdev

dragino committed:
Merge branch 'lg02' of github.com:dragino/dragino-packages into lg02

dragino committed:
fix bug of cac foptlen for decode payload

dragino committed:
update GUI related setting

dragino committed:
fix the len of payloadtxt

mikayong committed:
add support for custom downlink

mikayong committed:
fix RF Power adjusted

Commit:
fix txlut table

dragino committed:
update files

dragino committed:
updtae port filter

dragino committed:
fix bug in rf power control

mikayong committed:
fix bug

mikayong committed:
fix bug

dragino committed:
fix typo

mikayong committed:
add customer downlink support

mikayong committed:
Merge branch 'lg02' of github.com:dragino/dragino-packages into lg02

mikayong committed:
fix gps format error

dragino committed:
mosquitto client option -o for output subscribe message

mikayong committed:
add sx1302 test program

mikayong committed:
Merge branch 'lg02' of github.com:dragino/dragino-packages into lg02

dragino committed:
add sx1302 test prog

mikayong committed:
run reset_lgw script in main thread

mikayong committed:
reset_lgw fix

dragino committed:
update mqtt support

dragino committed:
add devskey file

dragino committed:
remove sx1302 package

mikayong committed:
remove mosquitto include from lg01&lg02

mikayong committed:
Merge branch 'lg02' of github.com:dragino/dragino-packages into lg02

dragino committed:
no reconnect for none-lorawan connection

dragino committed:
add back deleted content by mistake

mikayong committed:
modify 01/02 fwd push up

mikayong committed:
change fwd uplink deal

dragino committed:
Update

joerg555 committed:
Security Fix:
new jstrcpyn()
replace strcpy to jstrcpyn
fix stuff

joerg555 committed:
fix stuff

joerg555 committed:
fix start_time, end_time calculation

mikayong committed:
update for lbt

dragino committed:
update

dragino committed:
add RSSI and SNR i payload

joerg555 committed:
Merge remote-tracking branch 'remotes/dragino/lg02' into lg02

mikayong committed:
support for ftdi device FT2232D

dragino committed:
fix bug of previous update

dragino committed:
update , change please check the firmware changelog

dragino committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

joerg555 committed:
Merge branch 'lg02' of github.com:dragino/dragino-packages into lg02

dragino committed:
add new UI package

mikayong committed:
update

mikayong committed:
update

mikayong committed:
udpate makefile

mikayong committed:
update libftdi

dragino committed:
modify Makefile to remove unnecessary debug utility

dragino committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

mikayong committed:
fix ftdi device write interface

mikayong committed:
Merge branch 'lg02' of github.com:dragino/dragino-packages into lg02

dragino committed:
Merge pull request #4 from joerg555/lg02

Lg02 security fixes

dragino committed:
fix push_up data : rxdev->cr type

dragino committed:
Change UCI sx1301 file structure.

dragino committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

mikayong committed:
fix regional parameters

mikayong committed:
fix regional parameters

mikayong committed:
add relay mode

mikayong committed:
remove message display of not detect sx127x

mikayong committed:
add auto configure by script when plug usb disk

dragino committed:
update work space

dragino committed:
add config option(isftdi) in sx130x config files

mikayong committed:
logread format adjust

dragino committed:
logread format adjust

dragino committed:
fix the logread output format

mikayong committed:
add traffic fo gateway

mikayong committed:
update

mikayong committed:
Merge branch 'lg02' of github.com:dragino/dragino-packages into lg02

dragino committed:
Add SX1302. LIg16 support

dragino committed:
修改sqlite数据库路径

dragino committed:
add tcp process and gateway traffic

mikayong committed:
增加自定义下发的参数，可自定义窗口、频率、速率、带宽、功率

mikayong committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

mikayong committed:
improve LG01-N packt lost status

mikayong committed:
修改自定义下发的格式，可以自定义下发的LoRa参数

mikayong committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

mikayong committed:
db: 将db移到/tmp/db.sqlite, 启动从/etc/lora/devskey复制数据

mikayong committed:
分存数据库

mikayong committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

mikayong committed:
Fix INDIA ABP DOWNLINK Frequency

mikayong committed:
generate-config.sh:spell check, add option mac2db and custom_downlink

mikayong committed:
add file lora-filter.has

mikayong committed:
verify fcnt for abp decode

mikayong committed:
generate-config.sh: modify log_mask type, mac_decoded to mac_decode; lora_gw: link fwd

mikayong committed:
update source

mikayong committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

mikayong committed:
readd

mikayong committed:
remove conflict

mikayong committed:
lora_gw: station link to sx130x, station.conf copy from sx130x configure

mikayong committed:
add regional option

mikayong committed:
log_mask set to default value '1111'

mikayong committed:
fix bigendian byte order

mikayong committed:
add link form ssh to rssh for rssh connect

mikayong committed:
remove syble link

mikayong committed:
Update rssh_client , add MA869 band.
add TTNv3 server
fix LG02 to be able to receive 255 bytes

mikayong committed:
Support min freq on Helium

mikayong committed:
add chip detect script

mikayong committed:
Update to fix bugs. please check the change on the latest firmware release at 2021-Jun-20

mikayong committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

Qqwy commented:
I managed to download and build the project on an old VPS which still is running Ubuntu 14.04. On my current 'normal' machine, the version of `ncurses` is already so much newer that the `menuconfig`-program does not even start up :man_shrugging: .

You might be able to reproduce it by e.g. running version 14.04 of Ubuntu in a virtual machine or container.

The steps I followed were these:

2. Download and extract Dragino's OpenWRT SDK fork. Make it your current directory in the shell.
2. Run `./scripts/feeds update -a` (**This is missing from Dragino's the current readme about building your own package!**)
3. Copy the files [in this folder](https://github.com/dragino/dragino-packages/tree/lg02/lg02-pkt-fwd) over to `./package/lg02-pkt-fwd`.
4. Be sure to remove all current `*.o` object files which are in the `./package/lg02-pkt-fwd` directory! Even `make clean` will not correctly remove them (and instead might remove other files that the build process depends on, which are not remade when running a normal `make`, so then you'll have to restart from step (1.) ...)
3. Run `make menuconfig`, and be sure that the package is selected under 'Utilities'. Save this config and exit.
4. Make any changes to the source code you wish, by modifying the files at `./package/lg02-pkt-fwd/...`.
4. Run `make`. If there's errors, you can see them in detail by running `make -j1 V=s` instead.
5. If compilation was successful, the built package can now be found under `./bin/packages/mips_24kc/base/`. It is named something like `lg02_pkt_fwd_1.2.1-1_mips_24kc.ipk`.
6. Copy it over to your Dragino LG02, using e.g. `scp` (for instance: `scp -P 2222 ./lg02_pkt_fwd_1.2.1-1_mips_24kc.ipk root@dragino_ip_address:/tmp/` .
7. On the Dragino run `opkg remove lg02_pkt_fwd` followed by `opkg install /tmp/lg02_pkt_fwd_1.2.1-1_mips_24kc.ipk` (or what the exact filename may be).

For clarity: I wanted to make changes to the code of the Dragino LG02's `lg02_pkt_fwd` code. I expect that for one of the other packages the process will be similar, but you might need to make some further changes.

mikayong committed:
add dragino-gw-fwd applications

mikayong committed:
add debug_debug option for output debug info

mikayong committed:
adjust dnlink,remove duplicate devaddr package

mikayong committed:
fix dnlink queue problem

mikayong committed:
update

mikayong committed:
fix bug of dn_link

mikayong committed:
fix bug of dnlink

mikayong committed:
update

mikayong committed:
Update to lgw--build-v5.4.1640315898-20211224-1120

mikayong committed:
Update To latest firmware version

mikayong committed:
Add fwd code

mikayong committed:
Update Code to latest firmware

mikayong committed:
Update Source

mikayong committed:
update

mikayong committed:
update Source
add fwd_lbt
fix decoder
fix haserl-ui problem

mikayong committed:
Add cfg-302-zn
Add auto-opkg
Optimize aws configuration
Optimize cfg-302 parameters

mikayong committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

mikayong committed:
add config files for lora gateway

mikayong committed:
fix

mikayong committed:
Add fwd profile
Add decode file
Optimize fwd recycling threads
Add auto-update
Fix iot_keep_alive bug
Fix haserl-ui display bug

mikayong committed:
Merge branch 'lg02' of https://github.com/dragino/dragino-packages into lg02

mikayong committed:
Update Web GUI and gw-fwd

mikayong committed:
Add get imei file

mikayong committed:
20240109 updated log:
1. Optimizing the online detection of fwd
2. Fix the global file for MA869
3. Add Nwkid Filter
4. Add no network detection
...

mikayong committed:
Update

mikayong committed:
1. Added filter for fwd: Dev EUI,NetID
2. Added BasicStation filter
3. Added no network detection
4. Fixed some WebUI bugs
5. Optimized detection of IotKeepalive
